.github/workflows/ci.yml

```
@@ -1,181 +1,148 @@
 jobs:
-  test:
+  # Backend Testing and Linting
+  backend-test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11']
     
     steps:
-    - uses: actions/checkout@v4
+    - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)